charmed-osm/kafka-k8s-operator
# Add metrics support and integration with COS

- Add `metrics` configuration option
- Add integration with grafana-k8s
- Add integration with prometheus-k8s


Steps to test:

```shell
charmcraft pack
juju add-model kafka-k8s
juju deploy ./kafka-k8s_ubuntu-20.04-amd64.charm \
  --resource kafka-image=confluentinc/cp-kafka:7.0.1 \
  --resource jmx-prometheus-jar=./jmx_prometheus_javaagent-0.15.0.jar
juju deploy zookeeper-k8s --channel edge
juju relate kafka-k8s zookeeper-k8s
juju deploy cos-lite --channel beta --trust
juju upgrade-charm grafana --channel edge
juju relate grafana kafka-k8s
juju relate prometheus kafka-k8s
watch -c juju status --color  # wait
juju run-action grafana/0 get-admin-password --wait
juju status grafana
```

Then go to http://<grafana_ip>:3000, login with user `admin`.

Click to the search button, and Kafka dashboards should appear there.
